resources/views/platform/components/game/mature-content-gate.blade.php

```diff
@@ -0,0 +1,72 @@
+@props([
+    'gameId' => 0,
+    'gameTitle' => 'Unknown game',
+    'consoleId' => 0,
+    'consoleName' => 'Unknown console',
+    'userWebsitePrefs' => 0,
+])
+
+<?php
+use App\Site\Enums\UserPreference;
+
+$matureContentPref = UserPreference::Site_SuppressMatureContentWarning;
+?>
+
+<script>
+function matureContentNoticeComponent() {
+    return {
+        async disableMatureContentWarningPreference() {
+            const existingUserPrefs = {{ $userWebsitePrefs }} ?? 0;
+            const matureContentPref = {{ $matureContentPref }} ?? 0;
+
+            const newPreferencesValue = {{ ($userWebsitePrefs ?? 0) | (1 << $matureContentPref) }};
+            const gameId = {{ $gameId }};
```

Comment:
It actually looks like most of this code can be generated on the server. This is what gets output currently:
```
async disableMatureContentWarningPreference() {
   const existingUserPrefs = 8319 ?? 0;
   const matureContentPref = 7 ?? 0;
    
   const newPreferencesValue = 8447;
   const gameId = 17703;
    
   await fetcher('/request/user/update-preferences.php', {
     method: 'POST',
     body: `preferences=${newPreferencesValue}`
   });
    
   window.location = 'http://localhost/game/17703';
}
```

If you can move `newPreferencesValue` into the body, all the `const`s could be eliminated.

Reply:
Indeed, this can be much more concise. It also just occurred to me we could eliminate setting `window.location` to some string. Here's the function as of https://github.com/RetroAchievements/RAWeb/pull/1879/commits/0bbf678e41e635586d96f7110089f6bd0b7cda5c:

```js
async disableMatureContentWarningPreference() {
    await fetcher('/request/user/update-preferences.php', {
        method: 'POST',
        body: "preferences={{ $newPreferencesValue }}"
    });

    window.location.reload();
}
```